modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -1030,21 +1030,28 @@ static String matchingIntType(boolean unsigned, Long inclusiveMin, Long inclusiv
             Map<String, Object> mo = (Map<String, Object>) _mo;
             CodegenModel cm = (CodegenModel) mo.get("model");
 
-
             if (cm.dataType != null && cm.dataType.equals("object")) {
                 // Object isn't a sensible default. Instead, we set it to
                 // 'null'. This ensures that we treat this model as a struct
                 // with multiple parameters.
                 cm.dataType = null;
             } else if (cm.dataType != null) {
-                // We need to hack about with single-parameter models to get
-                // them recognised correctly.
-                cm.isAlias = false;
-                cm.dataType = typeMapping.get(cm.dataType);
+                if (cm.dataType == "map") {
```

Comment:
@bjgill should this be `"map".equals(cm.dataType)` instead?

Reply:
Good point - I'll be consistent with the equivalent 5 lines previously, though - `cm.dataType.equals("map")`